packages/app/src/systems/Transaction/services/transaction.tsx

```diff
@@ -98,7 +139,88 @@ export class TxService {
   }: TxInputs['simulateTransaction']) {
     const provider = new Provider(providerUrl || '');
     const { receipts } = await provider.call(transactionRequest);
-
     return receipts;
   }
+
+  static async createTransfer(input: TxInputs['createTransfer']) {
+    const request = new ScriptTransactionRequest({ gasPrice: 0 });
+    const to = Address.fromAddressOrString(input.to);
+    const { assetId, amount } = input;
+    request.addCoinOutput(to, amount, assetId);
+    return request;
+  }
+
+  static async getTxWithResources(input: TxInputs['getTxWithResources']) {
+    const { needToAddResources = true, gasFee = bn(0) } = input || {};
+    const request = transactionRequestify(input.tx);
+    const coins = request.getCoinOutputs();
+    const requiredCoins = coins.map((coin) => {
+      let amount = bn(coin.amount);
+      if (coin.assetId === NativeAssetId) {
+        amount = amount.add(gasFee || bn(0));
+      }
+      return { assetId: coin.assetId, amount };
+    });
+
+    if (needToAddResources) {
+      const resources = await provider.getResourcesToSpend(
+        input.wallet.address,
+        requiredCoins
+      );
+      request.addResources(resources);
+    }
+    return request;
+  }
+
+  static async fundTransaction(input: TxInputs['fundTransaction']) {
+    const { wallet, tx } = input;
+    const preRequest = await TxService.getTxWithResources({ wallet, tx });
+    const txCost = await wallet.provider.getTransactionCost(preRequest);
```

Comment:
can we try to get the same `txCost` info, but using `tx utils` ? would be better because:

1- `getTransactionCost` is broken, because it uses `calculateTransactionFee`, which is using a fixed `GAS_PRICE_FACTOR`.
2- `tx utils` will become the default way to get this kind of info, and this provider methods will be deprecated